src/Async/asyncToPromise.js

```diff
@@ -0,0 +1,26 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis */
+
+const curry = require('../core/curry')
+const isSameType = require('../core/isSameType')
+const isFunction = require('../core/isFunction')
+
+const Async = require('../core/types').proxy('Async')
+
+const toPromise = m => {
+  if(!isSameType(Async, m)) {
+    throw new TypeError('asyncToPromise: Async or a function returning an Async required')
+  }
+
+  return m.toPromise()
+}
+
+// asyncToPromise :: Async e a -> Promise a e
+// asyncToPromise :: (a -> Async e b) -> a -> Promise b e
+function asyncToPromise(async) {
+  return isFunction(async) ?
+    x => toPromise(async(x)) :
```

Comment:
Mostly a matter of style, and have not found a rule yet in lint, but could you change this up to read:
```js
return isFunction(m)
  ? x => toProimse(m(x))
  : toPromise(m)
```